capsules/src/ble_advertising_driver.rs

```diff
@@ -836,19 +548,27 @@ where
         &self,
         command_num: usize,
         data: usize,
-        _: usize,
+        interval: usize,
         appid: kernel::AppId,
     ) -> ReturnCode {
         match command_num {
-            // Start periodic advertisments
+            // Start periodic advertisements
             0 => self.app
                 .enter(appid, |app, _| {
                     if let Some(BLEState::Initialized) = app.process_status {
-                        app.process_status = Some(BLEState::AdvertisingIdle);
-                        app.random_nonce = self.alarm.now();
-                        app.set_next_alarm::<A::Frequency>(self.alarm.now());
-                        self.reset_active_alarm();
-                        ReturnCode::SUCCESS
+                        let pdu_type = data as AdvPduType;
+                        match pdu_type {
+                            ADV_IND | ADV_NONCONN_IND | ADV_SCAN_IND => {
+                                app.pdu_type = pdu_type;
+                                app.process_status = Some(BLEState::AdvertisingIdle);
+                                app.random_nonce = self.alarm.now();
+                                app.advertisement_interval_ms = cmp::max(20, interval as u32);
```

Comment:
Should advertisement intervals less than 20 ms result in an error code instead? I don't know if it really matters.